### PR TITLE
Implemented and Tested Clear.

### PR DESCRIPTION
**Pull request by matt-tsai**
Implemented and tested Clear. Fixed some code spacing. Clear will be helping in testing other functions so more comprehensive tests will come with the migration of more features.

### CODE REVIEW

File: direct/tpm2/clear_test.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	}
+	defer thetpm.Close()
+
+	clear := Clear{
+		AuthHandle: tpm.RHLockout,
+	}
+
+	if _, err := clear.Execute(thetpm); err != nil {
+		t.Fatalf("Clear failed: %v", err)
+	}
```
Comment (josephlr) [inline direct/tpm2/clear_test.go:24]: Is it possible to have some observable affect here? I know TPM2_Clear does multiple things, would it be possible to observe one of them?

Comment (chrisfenner) [reply direct/tpm2/clear_test.go:24]: You could generate a primary key in the storage hierarchy with the same template before and after the clear, and assert the public part changed.

Comment (josephlr) [reply direct/tpm2/clear_test.go:24]: Good idea, I like that

**Comment by chrisfenner** [comment]
Added this with the requested test in #305

**Pull request closed without merging**